### ISSUE TITLE
Masking screws up model.evaluate

### ISSUE THREAD

**Issue by IchBinDerBoss** [keras-team/keras#11625]
Please make sure that the boxes below are checked before you submit your issue.
If your issue is an **implementation question**, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [on the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) instead of opening a GitHub issue.

Thank you!

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
`pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps`

- [?] Check that your version of TensorFlow is up-to-date. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).
-------------------------------------------------------------------------------------------------------------------
Code example: https://github.com/IchBinDerBoss/keras_bug

I trained a model to discriminate between five classes, using a softmax activation at the last layer and cross_entropy loss function. I removed most parts to have a small runnable version which I provide at the link above.

keras model.evaluate() function outputs NaNs for accuracy and loss in some cases even though model.predict outputs valid predictions and the true lables are also okay.

For the 10 datapoints I provide, it outputs NaN 7 times. Also it does not count them as wrong, but simply ignores those samples which in my case yields a too optimistic accuracy of 0.66 when 0.5 would have been appropriate.

After some testing, I found out, that this only happens when a Masking layer is applied to the input.
Still there is always one Input which is never masked away, so the model should be able to compute everything.

**Comment by IchBinDerBoss** [comment]
I just wanted to add, that this bug usually leads to a overestimation of the accuracy and other measures. 
For instance I had a test set, where model.evaluate() outputted an accuracy around 96% whereas the true accuracy was more in the region of 65%.
This might be problematic if one is relying on model.evaluate() printing the correct accuracy.
